lib/ecto/postgres/pg_rational.ex

```diff
@@ -0,0 +1,157 @@
+use Vtc.Ecto.Postgres.Utils
+
+defpgmodule Vtc.Ecto.Postgres.PgRational do
+  @moduledoc """
+  Defines a composite type for storing rational values as dual int64s. These values
+  are cast to `%Ratio()` structs for use in application code, provided by the `Ratio`
+  library.
+
+  The composite type is defined as follows:
+
+  ```sql
+  CREATE TYPE public.rational as (
+    numerator bigint,
+    denominator bigint
+  )
+  ```
+
+  Rational values can be intatiated in raw SQL queries like so:
+
+  ```sql
+  SELECT (1, 2)::rational
+  ```
+
+  ## Field migrations
+
+  You can create a field as a rational during a migration like so:
+
+  ```elixir
+  create table("rationals") do
+    add(:a, PgRational.type())
+    add(:b, PgRational.type())
+  end
+  ```
+
+  ## Schema fields
+
+  Then in your schema module:
+
+  ```elixir
+  defmodule MyApp.Rationals do
+  @moduledoc false
+  use Ecto.Schema
+
+  alias Vtc.Ecto.Postgres.PgRational
+
+  @type t() :: %__MODULE__{
+          a: Ratio.t(),
+          b: Ratio.t()
+        }
+
+  schema "rationals_01" do
+    field(:a, PgRational)
+    field(:b, PgRational)
+  end
+  ```
+
+  ... notice that the schema field type is `PgRational`, but the type-spec field uses
+  `Ratio.t()`, the type that our DB fields will be deserialized into.
+
+  ## Changesets
+
+  With the above setup, changesets should just work:
+
+  ```elixir
+  def changeset(schema, attrs) do
+    schema
+    |> Changeset.cast(attrs, [:a, :b])
+    |> Changeset.validate_required([:a, :b])
+  end
+  ```
+
+  Rational values can be cast from the following values in changesets:
```

Comment:
Since this function is mostly here to support casting from JSON, and since JSON doesn't deserialize to tuples, I don't think we need to worry about that yeah? Seems like something we leave out initially and then add is some user wants that for a real-world use case.

I only hesitate -- since this is a lib -- since I often end up adding more features than there is immediate call for since it seems like there COULD be a need for it, and then I'm stuck supporting it even if it's never used.